### PR TITLE
CVE-2017-5650

### PR DESCRIPTION
**Pull request by moiseslorap**
Apache Tomcat - DoS vulnerability due to mishandled processes of HTTP Requests.

### CODE REVIEW

File: cves/CVE-2017-5650.yml [hunk 1]
```diff
@@ -44,6 +44,11 @@ description_instructions: |
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
 description:
+  This CVE describes a Denial of Service attack done through the network via malicious HTTP requests.
+  More specifically, the flaw can be encountered in the handling of a HTTP/2 GOAWAY frame for a connection did not
+  close its associated streams that were waiting for a WINDOW_UPDATE before allowing the application to write more data.
+  These waiting streams each consumed a thread. Therefore, an attacker could generate a DoS attack through a malicious client
+  that constructed a series of HTTP/2 requests that would consume all available processing threads.
```
Comment (hmeinertrita) [inline cves/CVE-2017-5650.yml]: There is a lot of jargon here that makes this hard to read. Things like "HTTP/2 GOAWAY frame" and "WINDOW_UPDATE" make this hard to understand for someone with only general programming knowledge, who this description should be written for.

File: cves/CVE-2017-5650.yml [hunk 2]
```diff
@@ -101,9 +106,9 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
-  code:
-  fix:
+  answer: No unit tests were involved in the fix for this vulnerability.
```
Comment (hmeinertrita) [inline cves/CVE-2017-5650.yml]: I would elaborate more here. Did you find unit tests of the vulnerable code? If not, where did you look? If yes, why weren't they involved in the fix? What did the fix involve that didn't require changing existing unit tests/adding new ones?

File: cves/CVE-2017-5650.yml (hunk 1 above)
Comment (mll8657) [inline cves/CVE-2017-5650.yml]: Seconding hmeinertrita's comment and adding that, as someone who was told by professor Meneely to explain what an http header was, you could go further than just that jargon. Explaining processing threads and waiting streams consuming them would also be helpful.

File: cves/CVE-2017-5650.yml [hunk 3]
```diff
@@ -119,9 +124,11 @@ discovered:
     may leave the entries blank except for "answer", BUT please write down
     where you looked in "answer".
   answer:
-  date:
+    Chun Han Hsiao reported this vulnerability where he discovered that a remote user can consume excessive resources on the target system by
+    sending a series of specially crafted HTTP/2 requests to trigger a flaw in the processing of HTTP/2 GOAWAY frames and consume all available processing threads on the target system.
```
Comment (mll8657) [inline cves/CVE-2017-5650.yml]: I have to horizontally scroll to read these lines. Splitting them up would make them easier to read.

**Comment by mll8657** [comment]
+1 upvotes

File: cves/CVE-2017-5650.yml (hunk 1 above)
Comment (moiseslorap) [reply cves/CVE-2017-5650.yml]: Gave a more elaborate description.

File: cves/CVE-2017-5650.yml (hunk 2 above)
Comment (moiseslorap) [reply cves/CVE-2017-5650.yml]: Elaborated on why no unit tests are necessary.

File: cves/CVE-2017-5650.yml (hunk 3 above)
Comment (moiseslorap) [reply cves/CVE-2017-5650.yml]: fixed

**Pull request merged**